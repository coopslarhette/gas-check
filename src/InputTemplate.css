.input-group {
    margin-top: 20px;
    margin-bottom: 0!important;
}

@media (min-width: 768px) {
    .input-group {
        margin-left: 30px;
    }
}
