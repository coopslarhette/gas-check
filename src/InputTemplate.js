import React from 'react'
import InputGroup from 'react-bootstrap/InputGroup'
import FormControl from 'react-bootstrap/FormControl'


function InputTemplate(props) {
  const { placeholder, prepend } = props
  return (
    <div>
      <InputGroup className="mb-3">
        <InputGroup.Prepend>
          <InputGroup.Text id="basic-addon1">{prepend}</InputGroup.Text>
        </InputGroup.Prepend>
        <FormControl
          placeholder={placeholder}
          aria-label="Input"
          aria-describedby="basic-addon1"
        />
      </InputGroup>
    </div>
  )
}

export default InputTemplate
