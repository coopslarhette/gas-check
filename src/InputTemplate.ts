import React from 'react'
import InputGroup from 'react-bootstrap/InputGroup'
import FormControl from 'react-bootstrap/FormControl'
import './InputTemplate.css'

function InputTemplate(props) {
  const {
    // TODO maybe switch to TS to avoid this
    // eslint-disable-next-line react/prop-types
    placeholder, prepend, validationRegex, formIdentifier,
  } = props
  let inputFormNode

  return (
    <InputGroup className="mb-3 input-group">
      <InputGroup.Prepend>
        <InputGroup.Text id="basic-addon1">{prepend}</InputGroup.Text>
      </InputGroup.Prepend>
      <FormControl
        placeholder={placeholder}
        aria-label="Input"
        aria-describedby="basic-addon1"
        ref={(ref) => {
          inputFormNode = ref
        }}
        onChange={() => {
          // eslint-disable-next-line react/prop-types
          props.isValidInput(formIdentifier, validationRegex.test(inputFormNode.value))
        }}
      />
    </InputGroup>
  )
}

export default InputTemplate
