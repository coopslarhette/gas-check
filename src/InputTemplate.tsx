import React from 'react'
import InputGroup from 'react-bootstrap/InputGroup'
import FormControl from 'react-bootstrap/FormControl'
import './InputTemplate.css'

type Props = {
  placeholder: string
  prepend: string
  validationRegex: RegExp
  formIdentifier: string
  validateChange: (formIdentifier: string, isValid: boolean) => void
  storeInputValue: (formValue: string, formIdentifier: string) => void
}

function InputTemplate(props: Props): JSX.Element {
  const {
    placeholder, prepend, validationRegex, formIdentifier,
  } = props
  let inputFormNode

  return (
    <InputGroup className="mb-3">
      <InputGroup.Prepend>
        <InputGroup.Text id="basic-addon1">{prepend}</InputGroup.Text>
      </InputGroup.Prepend>
      <FormControl
        placeholder={placeholder}
        aria-label="Input"
        aria-describedby="basic-addon1"
        ref={(ref) => {
          inputFormNode = ref
        }}
        onChange={(): void => {
          props.validateChange(formIdentifier, validationRegex.test(inputFormNode.value))
          props.storeInputValue(inputFormNode.value, formIdentifier)
        }}
      />
    </InputGroup>
  )
}

export default InputTemplate
