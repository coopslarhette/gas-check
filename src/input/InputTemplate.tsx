import React from 'react'
import InputGroup from 'react-bootstrap/InputGroup'
import FormControl from 'react-bootstrap/FormControl'
import './InputTemplate.css'

type PropTypes = {
  placeholder: string; prepend: string; validationRegex: RegExp; formIdentifier: string;
  validateChange: (formIdentifier: string, isValid: boolean) => void;
  storeInputValue: (formValue: string, formIdentifier: string) => void;
}

function InputTemplate({
  placeholder, prepend, validationRegex, formIdentifier, validateChange, storeInputValue,
}: PropTypes): JSX.Element {
  let inputFormNode

  return (
    <InputGroup className="mb-3">
      <InputGroup.Prepend>
        <InputGroup.Text id="basic-addon1">{prepend}</InputGroup.Text>
      </InputGroup.Prepend>
      <FormControl
        placeholder={placeholder}
        aria-label="Input"
        aria-describedby="basic-addon1"
        ref={(ref) => {
          inputFormNode = ref
        }}
        onChange={(): void => {
          validateChange(formIdentifier, validationRegex.test(inputFormNode.value))
          storeInputValue(inputFormNode.value, formIdentifier)
        }}
      />
    </InputGroup>
  )
}

export default InputTemplate
