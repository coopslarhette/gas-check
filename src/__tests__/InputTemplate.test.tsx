import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'
import InputTemplate from '../InputTemplate'

describe('Input template component', () => {
  let validateChange: (string, boolean) => void
  let validity: boolean
  beforeEach(() => {
    validateChange = (_: string, isValid: boolean): void => {
      validity = isValid
    }
  })

  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<InputTemplate
      placeholder="Test"
      prepend="A"
      isValidInput={validateChange}
      validationRegex={/.*/}
      formIdentifier="test"
    />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders all parts correctly', () => {
    const tree = renderer
      .create(<InputTemplate
        placeholder="Test"
        prepend="A"
        isValidInput={validateChange}
        validationRegex={/.*/}
        formIdentifier="test"
      />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('calls on function in isValidInput with a false boolean when input is empty', () => {
    const div = document.createElement('div')
    ReactDOM.render(<InputTemplate
      placeholder="Test"
      prepend="A"
      isValidInput={validateChange}
      validationRegex={/.+/}
      formIdentifier="test"
    />, div)

    expect(validity).toBeFalsy()
    ReactDOM.unmountComponentAtNode(div)
  })

  const setup = () => {
    const utils = render(<InputTemplate
      placeholder="Test"
      prepend="A"
      isValidInput={validateChange}
      validationRegex={/test/}
      formIdentifier="test"
    />)
    console.log(utils.baseElement)
    return utils
  }

  // it('calls on function in isValidInput with a true boolean when input is valid', () => {
  //   // const input = setup()
  //   const div = document.createElement('div')
  //   ReactDOM.render(<InputTemplate
  //     placeholder="Test"
  //     prepend="A"
  //     isValidInput={validateChange}
  //     validationRegex={/.+/}
  //     formIdentifier="test"
  //   />, div)
  //
  //   console.log(div)
  //   // fireEvent.change(input, { target: { value: '23' } })
  //   expect(validity).toBeTruthy()
  // })
})
