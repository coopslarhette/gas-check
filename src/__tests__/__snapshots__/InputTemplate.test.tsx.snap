// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input template component renders all parts correctly 1`] = `
<div
  className="mb-3 input-group input-group"
>
  <div
    className="input-group-prepend"
  >
    <span
      className="input-group-text"
      id="basic-addon1"
    >
      A
    </span>
  </div>
  <input
    aria-describedby="basic-addon1"
    aria-label="Input"
    className="form-control"
    onChange={[Function]}
    placeholder="Test"
  />
</div>
`;
